backend/layers/processing/process_validate_atac.py

```diff
@@ -0,0 +1,224 @@
+            else:  # fragment file optional
+                self.logger.info("Fragment is optional and not present. Skipping fragment validation.")
+                self.update_processing_status(
+                    dataset_version_id,
+                    DatasetStatusKey.ATAC_FRAGMENT,
+                    DatasetConversionStatus.SKIPPED,
+                    validation_errors=["Fragment is optional and not present."],
+                )
+                return True
+        return False
+
+    def hash_file(self, file_name: str) -> str:
+        """
+        Hash the file
+        :param file_name: the file to hash
+        :return: the hash
+        """
+        hashobj = hashlib.md5()
+        buffer = bytearray(2**18)
+        view = memoryview(buffer)
+        with open(file_name, "rb") as f:
+            while chunk := f.readinto(buffer):
+                hashobj.update(view[:chunk])
+        return hashobj.hexdigest()
+
+    def process(
+        self,
+        collection_version_id: CollectionVersionId,
+        dataset_version_id: DatasetVersionId,
+        manifest: IngestionManifest,
+        datasets_bucket: str,
+    ):
+        """
+
+        :param collection_version_id:
+        :param dataset_version_id:
+        :param manifest:
+        :param datasets_bucket:
+        :return:
+        """
+        # Download the original dataset files from URI
+        local_anndata_filename = self.download_from_source_uri(
+            source_uri=str(manifest.anndata),
+            local_path=CorporaConstants.ORIGINAL_H5AD_ARTIFACT_FILENAME,
+        )
+
+        if self.skip_atac_validation(local_anndata_filename, manifest, dataset_version_id):
+            return
+
+        # Download the original fragment file from URI
+        local_fragment_filename = self.download_from_source_uri(
+            source_uri=str(manifest.atac_fragment), local_path=CorporaConstants.ORIGINAL_ATAC_FRAGMENT_FILENAME
+        )
+
+        # Hash the original fragment file. This is used to check if the new fragment file is the same as the old
+        # fragment file to avoid uploading the same file multiple times
+        if self.business_logic.is_already_ingested(manifest.atac_fragment):
+            original_fragment_hash = self.hash_file(local_fragment_filename)
+        else:
+            # new fragment file is from a private source, so we must upload it
+            original_fragment_hash = None
+
+        # Validate the fragment with anndata file
+        try:
+            errors = self.schema_validator.validate_atac(local_fragment_filename, local_anndata_filename)
+        except Exception as e:
+            # for unexpected errors, log the exception and raise a ValidationAtacFailed exception
+            self.logger.exception("validation failed")
+            self.update_processing_status(
+                dataset_version_id, DatasetStatusKey.ATAC_FRAGMENT, DatasetConversionStatus.FAILED
```

Comment:
we should make ValidationStatus FAILED as well, right?

Reply:
That will happen here
https://github.com/chanzuckerberg/single-cell-data-portal/blob/460bf988b5206454c17e22d672ed49ed9fe8c6c9/backend/layers/processing/process.py#L146-L146